site-src/v1alpha2/guides/multiple-ns.md

```diff
@@ -1,158 +1,163 @@
-This allows the certificate to be managed centrally for all applications which 
-are using this Gateway.
-- It allows any Route in the cluster to use this Gateway because `namespaces.from = All`. 
-This is a permissive method of Route selection since the Routes are given 
-full control to select this Gateway. There are more restrictive forms of Route 
-selection that allow selection on a per-Namespace basis, detailed 
-in [Route binding](/concepts/api-overview/#route-binding). The following block
-specifies how this Gateway allows HTTPRoutes from all Namespaces in the 
-cluster to bind to it:
+The `https` listener in the above Gateway matches traffic for the
+`foo.example.com` domain. This allows the infrastructure team to manage all 
+aspects of the domain. The HTTPRoutes below do not need to specify domains
+and will match all traffic by default if `hostname` is not set. This makes
+it easier to manage HTTPRoutes because they can be domain agnostic, which is
+helpful when application domains are not static.
+
+This Gateway also configures HTTPS using the `foo-example-com` Secret
+in the `infra-ns` Namespace. This allows the infrastructure team to centrally
+manage TLS on behalf of app owners. The `foo-example-com` certificate will
+terminate all traffic going to all its attached Routes, without any TLS configuration
+on the HTTPRoutes themselves.
+
+This Gateway uses attachment constraints to specify the Namespaces from which
+HTTPRoutes can attach. This allows the infrastructure team to constrain who
+or which apps can use this Gateway by allowlisting the set of Namespaces.
+
 
 ```yaml
-    routes:
-      kind: HTTPRoute
+...
+  listeners:
+  - routes:
       namespaces:
-        from: "All"
+        selector:
+          matchLabels:
+          	shared-gateway-access: "true"
+...
 ```
 
-Meanwhile, the store team deploys their route for the `store` Service in the 
-`store-ns` Namespace:
+_Only_ Namespaces which are labelled `shared-gateway-access: "true"` will be able to attach
+their Routes to `shared-gateway`. In the following set of Namespaces,
+if an HTTPRoute existed in the `no-external-access` Namespace with a `parentRef` 
+for `infra-ns/shared-gateway`, it would be ignored by the Gateway because the 
+attachment constraint (Namespace label) was not met.
 
 ```yaml
-{% include 'v1alpha1/cross-namespace-routing/store-route.yaml' %}
+{% include 'v1alpha2/cross-namespace-routing/0-namespaces.yaml' %}
 ```
 
-This Route has straightforward routing logic as it just matches for 
-`/store` traffic which it sends to the `store` Service. The following snippet 
-of the [`gateways` field](/v1alpha2/references/spec/#networking.x-k8s.io/v1alpha1.RouteGateways) 
-controls which Gateways this Route can bind to:
+Note that attachment constraints on the Gateway are not required, but they are 
+ a best-practice if operating a cluster with many different teams
+and Namespaces. In environments where all apps in a cluster have permission to
```

Comment:
A small nit that applies a few places in this file - there are some inconsistent line lengths and some extra leading spaces that would be nice to remove.